server/src/main/java/org/elasticsearch/search/dfs/DfsSearchResult.java

```diff
@@ -138,9 +138,14 @@ public static void writeTermStats(StreamOutput out, TermStatistics[] termStatist
     }
 
     public  static void writeSingleTermStats(StreamOutput out, TermStatistics termStatistic) throws IOException {
-        assert termStatistic.docFreq() >= 0;
-        out.writeVLong(termStatistic.docFreq());
-        out.writeVLong(addOne(termStatistic.totalTermFreq()));
+        if (termStatistic != null) {
+            assert termStatistic.docFreq() > 0;
+            out.writeVLong(termStatistic.docFreq());
+            out.writeVLong(addOne(termStatistic.totalTermFreq()));
```

Comment:
let's not add one anymore for indices on or after 7.0?

Reply:
agreed, I pushed https://github.com/elastic/elasticsearch/pull/33310/commits/647b846d21c55a72dae0a512ada1dde15ecbfa26